packages/blade/src/components/Amount/Amount.tsx

```diff
@@ -0,0 +1,271 @@
+import type { ReactElement } from 'react';
+import type { FontSize } from '../../tokens/global/typography';
+import type { HeadingProps, TextProps, TitleProps } from '../Typography';
+import {
+  amountTextSizes,
+  currencyAbbreviationsMapping,
+  currencyLocaleMapping,
+  currencyPrefixMapping,
+  affixFontSizes,
+} from './amountTokens';
+import { BaseText } from '~components/Typography/BaseText';
+import type { Feedback } from '~tokens/theme/theme';
+import type { BaseTextProps } from '~components/Typography/BaseText/types';
+import BaseBox from '~components/Box/BaseBox';
+import type { TestID } from '~src/_helpers/types';
+import { metaAttribute, MetaConstants } from '~utils';
+
+type Currency = 'INR' | 'MYR';
+
+type AmountProps = {
+  /**
+   * The value to be rendered within the component.
+   *
+   */
+  value: number;
+  /**
+   * Sets the intent of the amount.
+   *
+   * @default undefined
+   */
+  intent?: Exclude<Feedback, 'neutral'> | undefined;
+  /**
+   * Sets the size of the amount.
+   *
+   * @default 'heading'
+   */
+  size?:
+    | `title-${NonNullable<Exclude<TitleProps['size'], 'large'>>}`
+    | `heading-${NonNullable<Exclude<HeadingProps<{ variant: 'regular' }>['size'], 'medium'>>}`
+    | `heading-${NonNullable<
+        Exclude<HeadingProps<{ variant: 'regular' }>['size'], 'medium'>
+      >}-${NonNullable<Exclude<TextProps<{ variant: 'body' }>['weight'], 'regular'>>}`
+    | `body-${NonNullable<Exclude<TextProps<{ variant: 'body' }>['size'], 'xsmall'>>}`
+    | `body-${NonNullable<Exclude<TextProps<{ variant: 'body' }>['size'], 'xsmall'>>}-${NonNullable<
+        Exclude<TextProps<{ variant: 'body' }>['weight'], 'regular'>
+      >}`;
+  /**
+   * Indicates what the suffix of amount should be
+   *
+   * @default 'decimals'
+   */
+  suffix?: 'decimals' | 'none' | 'humanize';
+  /**
+   * Highlights the main amount by making the prefix symbol and decimal digits subtle
+   *
+   * @default true
+   */
+  isAffixSubtle?: true | false;
+  /**
+   * Prefix to be shown before the amount value. The prefix can be either a currency symbol or a currency code.
+   *
+   * @default 'currency-symbol'
+   */
+  prefix?: 'currency-symbol' | 'currency-code';
+} & TestID;
+
+type ColorProps = {
+  amountValueColor: BaseTextProps['color'];
+  affixColor: BaseTextProps['color'];
+};
+
+const getTextColorProps = ({ intent }: { intent: AmountProps['intent'] }): ColorProps => {
+  const props: ColorProps = {
+    amountValueColor: 'surface.text.normal.lowContrast',
+    affixColor: 'surface.text.muted.lowContrast',
+  };
+  if (!intent) return props;
+  props.amountValueColor = `feedback.text.${intent}.lowContrast`;
+  props.affixColor = `feedback.text.${intent}.lowContrast`;
+  return props;
+};
+
+const getAffixFontWeight = (isAffixSubtle: true | false): 'regular' | 'bold' => {
+  if (isAffixSubtle) return 'regular';
+  return 'bold';
+};
+
+const getMainValueWeight = (
```

Comment:
Make the name more contextual?
```suggestion
const getAmountValueWeight = (
```